File: backend/backend/src/main/java/edu/cit/portes/studysessionscheduler/model/StudySession.java
package edu.cit.portes.studysessionscheduler.model;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Data
public class StudySession {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String topic;
    private String location;
    private String date;
    private String time;
    private int maxParticipants; // To prevent overcrowding
    private int currentParticipants = 0;
}

File: mobile/app/src/main/java/com/portes/it342mobile/StudySession.kt
package com.portes.it342mobile

data class StudySession(
    val id: Long? = null,
    val topic: String,
    val location: String,
    val date: String,
    val time: String,
    // Ensure these match the Spring Boot variable names exactly
    val maxParticipants: Int,
    val currentParticipants: Int = 0
)